### ISSUE TITLE
Add instructions for using private npm registries with Pages

### ISSUE THREAD

**Issue by caass** [cloudflare/cloudflare-docs#942]
@signalnerve 

1. Create and check in a `.npmrc` file with your registry ([details](https://docs.npmjs.com/cli/v7/configuring-npm/npmrc))
2. If using an environment variable (which are automatically substituted in `.npmrc` files), set that variable in your deployment settings (or wherever the tab is for setting environment variables)

e.g.

.npmrc:
```
@myscope:registry=https://npm.myprivateregistry.com/
//npm.myprivateregistry.com/:_authToken=${MY_AUTH_TOKEN}
```

environment variable config:
| Key           | Value        |
|---------------|--------------|
| MY_AUTH_TOKEN | XXX-XXX-XX-X |

**Comment by atymic** [comment]
I had a horrible exp with this. I don't want to `//npm.myprivateregistry.com/:_authToken=${MY_AUTH_TOKEN}` in my repo .`npmrc` as it means I can't use `yarn install` locally without setting the .env. 

I tried setting a pre install command that writes it to the file, but that doesn't work :(

**Comment by caass** [comment]
@atymic the workaround I used was to install [direnv](https://direnv.net/) and create a `.envrc` file that exported my auth token, maybe that will work for you as well?

**Comment by miguelcobain** [comment]
The problem with committing a `//npm.myprivateregistry.com/:_authToken=${MY_AUTH_TOKEN}` to the repo is that it forces every developer to have a `MY_AUTH_TOKEN` env variable locally.

@atymic I also tried the `preinstall` script trick to write to `.npmrc` file or to run the equivalent `npm config set` commands.
Neither worked. It's as if the configs/`.npmrc` changes are ignored for some reason.

**Comment by atymic** [comment]
Yep, I have the token in my home directory `.npmrc` but the local one im required to have for this repo overrides it

> @atymic I also tried the `preinstall` script trick to write to `.npmrc` file or to run the equivalent `npm config set` commands.
> Neither worked. It's as if the configs/`.npmrc` changes are ignored for some reason.


Yea, I couldn't get it to work, spent ages. Works fine on netlify and vercel.
I think the issue is that yarn install runs before the build command as well, so I can't even add a hack to override the default yarn install.

**Comment by miguelcobain** [comment]
@atymic in my case, I use npm, and preinstall **does** run before the build's `npm install`.
However, it's as if the changes are not affected.
Maybe because `npm install` is the thing that runs the preinstall, and so you can't change the config in the middle of the command?

**Comment by mckalexee** [comment]
I have similar issues. If there was any way to configure the npm config before the install that would be great.

```
//npm.myprivateregistry.com/:_authToken=${MY_AUTH_TOKEN}
```

just wont work due to the fact that it essentially breaks the ability to run `npm install` locally.

**Comment by miguelcobain** [comment]
We should reopen this ticket as it was not fully resolved by #1251

**Comment by FranRom** [comment]
Facing the same problem with Github package registry with a private package...
